state/dashboard/operations/searchMovies.js

```diff
@@ -3,34 +3,37 @@ import { normalize, schema } from 'normalizr';
 
 import { SEARCH_MOVIES } from '../types';
 import { mergeMoviesList } from '../../data/actions';
-import { fetchTrendingMoviesSuccess } from '../actions';
+import {
+  fetchMoviesSuccess,
+  fetchMoviesError,
+  setSearchMoviesQuery
+} from '../actions';
 
 const searchMoviesLogic = createLogic({
   type: SEARCH_MOVIES,
   warnTimeout: 0,
 
-  async process({ httpClient, action }, dispatch, done) {
-    const {
-      search,
-      setErrors,
-      setSubmitting,
-      page
-    } = action;
+  async process({ getState, httpClient, action }, dispatch, done) {
+    const { searchQuery, page } = action;
     const currentPage = page;
+    const query = searchQuery || getState().dashboard.searchQuery;
+
     try {
-      const { data } = await httpClient.get(`search/movie?query=${search}&page=${page}`);
+      const { data } = await httpClient.get(`search/movie?query=${query}&page=${page}`);
       // normalize data
       const movie = new schema.Entity('movies');
       const moviesSchema = { results: [movie] };
       const normalizedData = normalize(data, moviesSchema);
-      const { movies } = normalizedData.entities;
-      const ids = normalizedData.result.results;
+      const movies = normalizedData.entities.movies || {};
```

Comment:
Is it required to use empty object as fallback?

Reply:
Yes it is. Replaced condition to actions.